deploy-agent/deployd/client/client.py

```diff
@@ -44,8 +44,23 @@ def __init__(self, config=None, hostname=None, ip=None, hostgroup=None,
         self._autoscaling_group = None
         self._availability_zone = None
         self._stage_type = None
+        self._host_info_populated = False
+        try:
+            if self._read_host_info() is False:
```

Comment:
instead you can raise before `return False` below and get rid of the duplicated code, since it will hit the exception flow below. You can add a noop exception class above:
```
class HostReadFailedException(Exception):
    pass
```